packages/elements/src/chart/elements/chart.ts

```diff
@@ -0,0 +1,603 @@
+    super.updated(changedProperties);
+    if (changedProperties.has('config')) {
+      this.onConfigChange();
+    }
+  }
+
+  /**
+   * Element connected
+   * @returns {void}
+   */
+  public connectedCallback (): void {
+    super.connectedCallback();
+    this.setGlobalConfig();
+    if (this.canvas.value) {
+      this.createChart();
+    }
+  }
+
+  /**
+   * Element disconnected
+   * @returns {void}
+   */
+  public disconnectedCallback (): void {
+    super.disconnectedCallback();
+    this.destroyChart();
+  }
+
+  /**
+   * Create plugin to set our theme into ChartJS lifecycle
+   * @returns Created plugin
+   */
+  private createPlugin (): Plugin {
+    return {
+      id: 'ef-chart',
+      beforeInit: (chart: ChartJS) => {
+        const option: ChartOptions = this.themableChartOption;
+        merge(chart.config.options as ChartOptions, option);
+      },
+      beforeUpdate: this.decorateColors
+    };
+  }
+
+  /**
+   * Themable parts of the config.
+   * This will be merged into the configuration object.
+   * @returns Chart option with theme
+   */
+  protected get themableChartOption (): ChartOptions {
+    const boxWidth = this.cssVarAsNumber('--legend-key-box-width', '10') as number;
+    let boxHeight = Number(getComputedStyle(this).getPropertyValue('font-size').replace('px', ''));
+    if (this.config?.options?.plugins?.legend?.labels?.usePointStyle) {
+      boxHeight = boxWidth;
+    }
+
+    return {
+      animation: {
+        duration: this.cssVarAsNumber('--animation-duration', '0')
+      },
+      elements: {
+        line: {
+          borderWidth: this.cssVarAsNumber('--line-width', '1'),
+          tension: this.cssVarAsNumber('--line-tension', '0.5')
+        }
+      },
+      plugins: {
+        tooltip: {
+          backgroundColor: this.getComputedVariable('--tooltip-background-color', 'transparent'),
+          titleColor: this.getComputedVariable('--tooltip-title-color', 'transparent'),
+          bodyColor: this.getComputedVariable('--tooltip-body-color', 'transparent'),
+          cornerRadius: this.cssVarAsNumber('--tooltip-border-radius', '0'),
+          caretSize: this.cssVarAsNumber('--tooltip-caret-size', '0'),
+          padding: {
+            x: this.cssVarAsNumber('--tooltip-padding-x', '--tooltip-padding', '0'),
+            y: this.cssVarAsNumber('--tooltip-padding-y', '--tooltip-padding', '0')
+          },
+          titleSpacing: this.cssVarAsNumber('--tooltip-title-spacing', '0'),
+          displayColors: false
+        },
+        legend: {
+          position: ['pie', 'doughnut'].includes(this.config?.type || '') ? 'right' : 'top',
+          labels: {
+            boxWidth,
+            boxHeight,
+            generateLabels: this.generateLegendLabels
+          }
+        }
+      }
+    };
+  }
+
+  /**
+   * Set global configuration of ChartJS
+   * @returns {void}
+   */
+  // TODO: Try and remove the need for global object modification.
```

Comment:
Is this TODO still valid ? If it is, it is worth trying out ?

Reply:
I tried and found that without global config cannot cover our style to creating a new custom feature, for example a new custom scale. And also with global config is easier readability.

Reply:
So that means this TODO is not valid anymore. Should we remove it ?

Reply:
You're right. I will remove it.